src/System.CommandLine/Binding/ModelDescriptor.cs

```diff
@@ -35,7 +35,7 @@ protected ModelDescriptor(Type modelType)
         public IReadOnlyList<IValueDescriptor> PropertyDescriptors =>
             _propertyDescriptors ??=
                 ModelType.GetProperties(CommonBindingFlags)
-                         .Where(p => p.CanWrite)
+                         .Where(p => p.CanWrite && p.SetMethod.IsPublic)
```

Comment:
Why limit it to only settings properties that have public setters?

Reply:
This is less surprising than if we were to set non-public members.